benchexec/tools/nitwit.py

```diff
@@ -48,30 +47,27 @@ def version(self, executable):
     def name(self):
         return "Nitwit"
 
-    def cmdline(self, executable, options, tasks, propertyfile=None, rlimits={}):
-        if propertyfile:
-            options = options + ["-p", propertyfile]
-        return [executable] + options + tasks
+    def cmdline(self, executable, options, task, rlimits):
+        if task.property_file:
+            options = options + ["-p", task.property_file]
+        return [executable] + options + list(task.input_files_or_identifier)
 
-    def determine_result(self, returncode, returnsignal, output, isTimeout):
+    def determine_result(self, run):
         """
         See README.md at https://github.com/moves-rwth/nitwit-validator for information
         about result codes.
-        @param returncode: code returned
-        @param returnsignal: signal, which terminated validator
-        @param output: the std output
         @return: status of validator after executing a run
         """
-        if returnsignal == 0 and (returncode == 0 or returncode == 245):
+        if run.exit_code.signal is None and (run.exit_code.value == 0 or run.exit_code.value == 245):
```

Comment:
The check for `signal is None` is no longer neccessary, `value` is only a number if `signal` is `None`.
```suggestion
        if run.exit_code.value in [0, 245]:
```